tests/models/whisper/test_modeling_whisper.py

```diff
@@ -3386,6 +3386,66 @@ def test_tiny_static_generation(self):
         # assert re-ordered generations match those from eager
         assert (eager_generated_ids[permutation_idx, :] == static_generated_ids).all()
 
+    @slow
+    def test_tiny_static_generation_long_form(self):
+        import torch._dynamo.config
+
+        # only permit 4 compilations: 2 prefill steps and 2 decoding steps (1 for each of conditioned/not conditioned)
+        torch._dynamo.config.cache_size_limit = 4
+
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny.en")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny.en")
+        model.to(torch_device)
+
+        dataset = load_dataset("distil-whisper/meanwhile", "default")["test"]
+        dataset = dataset.cast_column("audio", Audio(sampling_rate=16000))
+        input_speech = [audio["array"] for audio in dataset[2:4]["audio"]]
+
+        inputs = processor(
+            input_speech,
+            return_tensors="pt",
+            padding="longest",
+            truncation=False,
+            return_attention_mask=True,
+            sampling_rate=16_000,
+        )
+        inputs = inputs.to(torch_device)
```

Comment:
can we also test with left padding side?

Reply:
It doesn't make sense to left pad an audio input 😅 We drop out audios from the batch as soon as they've finished generation, so we need to right pad the audios and keep track of where the padding (silence) begins through the attention mask

Reply:
But we do allow left padding in the Raise value error, maybe it's only ofr the decoder input ids

Reply:
Ah I think there's some confusion 😉 This is not referring to the padding from the tokenizer, but rather the padding to apply to tensors within generation with fallback.

E.g. we pad the condition tokens on the left:
https://github.com/huggingface/transformers/blob/ef47d06e6a409e44da6e4702cf0bc78f854f7287/src/transformers/models/whisper/generation_whisper.py#L1663-L1671

And the final sequences on the right:
https://github.com/huggingface/transformers/blob/ef47d06e6a409e44da6e4702cf0bc78f854f7287/src/transformers/models/whisper/generation_whisper.py#L728-L730

The current slow test sets `condition_on_prev_tokens=True`, so we can be sure it includes both cases!

Reply:
Cool thanks, I was a bit lost indeed!

Reply:
No worries - thanks for the speedy review!